src/shogun/multiclass/KNNSolver.h

```diff
@@ -0,0 +1,114 @@
+/* This software is distributed under BSD 3-clause license (see LICENSE file).
+ *
+ * Copyright (c) 2012-2013 Sergey Lisitsyn
+ */
+
+#ifndef KNNSOLVER_H__
+#define KNNSOLVER_H__
+
+#include <shogun/lib/config.h>
+
+#include <shogun/lib/common.h>
+#include <shogun/io/SGIO.h>
+#include <shogun/distance/Distance.h>
+#include <shogun/machine/DistanceMachine.h>
+
+namespace shogun
+{
+/**
+ * The KNNSolver class is the virtual base class of BruteKNNSolcer, CoverTreeKNNSolver, KDTreeKNNsolver, LSHKNNSolver. 
+ * The KNNSolver class will include most of Variables which been used all the other Solver and methods been used to choose 
```

Comment:
If you want to say something here, mention important methods (just say what they are intended to do). On the other hand, if the methods are well documented, no need to say a lot (apart from that this is the base class)